### PR TITLE
add chrony

### PR DESCRIPTION
**Pull request by hadfl**

### CODE REVIEW

**Review by oetiker — APPROVED** [review]

File: build/chrony/build.sh [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+CPPFLAGS="-I$DEPROOT$PREFIX/include"
+LDFLAGS64+=" -L$DEPROOT$PREFIX/lib/$ISAPART64"
+addpath PKG_CONFIG_PATH64 $DEPROOT$PREFIX/lib/$ISAPART64/pkgconfig
+
+#########################################################################
+
+test_config() {
+    for flag in $EXPECTED_OPTIONS; do
+        egrep -s "$flag 1" config.h || logerr "$flag not set"
```
Comment (citrus-it) [inline build/chrony/build.sh:54]:
```suggestion
       $EGREP -s "$flag 1" config.h || logerr "$flag not set"
```

File: build/chrony/files/chrony.conf [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+# per million.  When chronyd starts, the system clock can be tuned
+# immediately so that it doesn't gain or lose any more time.  You
+# generally want this, so it is uncommented.
+
+driftfile /var/lib/chrony/drift
+
+# If you want to enable NTP authentication with symmetric keys, you will need
+# to uncomment the following line and edit the file to set up the keys.
+
+! keyfile /etc/chrony.keys
```
Comment (citrus-it) [inline build/chrony/files/chrony.conf:79]:
```suggestion
! keyfile /etc/inet/chrony.keys
```

File: build/chrony/files/chrony.conf [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+# 1. If you stop chronyd and restart it with the '-r' option (e.g. after
+# an upgrade), the old measurements will still be relevant when chronyd
+# is restarted.  This will reduce the time needed to get accurate
+# gain/loss measurements.
+#
+# 2. On Linux, if you use the RTC support and start chronyd with
+# '-r -s' on bootup, measurements from the last boot will still be
+# useful (the real time clock is used to 'flywheel' chronyd between
+# boots).
+#
```
Comment (citrus-it) [inline build/chrony/files/chrony.conf:109]: Remove block.

File: build/chrony/files/chrony.conf [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+
+#######################################################################
+### HARDWARE TIMESTAMPING
+# On Linux, if the network interface controller and its driver support
+# hardware timestamping, it can significantly improve the accuracy of
+# synchronisation. It can be enabled on specified interfaces only, or it
+# can be enabled on all interfaces that support it.
+
+! hwtimestamp eth0
+! hwtimestamp *
```
Comment (citrus-it) [inline build/chrony/files/chrony.conf:271]: Remove

File: build/chrony/files/chrony.conf [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+# /dev/rtc.  If it's accessed somewhere else on your system (e.g. you're
+# using devfs), uncomment and edit the following line.
+
+! rtcdevice /dev/misc/rtc
+
+# Alternatively, if not using the -s option, this directive can be used
+# to enable a mode in which the RTC is periodically set to the system
+# time, with no tracking of its drift.
+
+! rtcsync
```
Comment (citrus-it) [inline build/chrony/files/chrony.conf:306]: Remove.

**Review by citrus-it — CHANGES_REQUESTED** [review]
The open-vm-tools smf manifest will need adjusting too.

**Comment by tomww** [comment]
May I ask to have added --enable-ntp-signd
(from https://wiki.samba.org/index.php/Time_Synchronisation#With_chrony_2:)
"chrony >= 3.0 from https://chrony.tuxfamily.org, compiled with enabled signed ntp support (**--enable-ntp-signd** when building chrony)"

That would greatly help in Samba Active Domain (e.g. SFEsamba410.spec) to have Windows clients get correct time.
It works like this: NTP packages are signed by the samba controller and sent to the clients. The socket is the connection between chrony and samba.

The path for the socket could be  /var/lib/samba/ntp_signd

Thank you!

**Comment by citrus-it** [comment]
> May I ask to have added --enable-ntp-signd

It looks straightforward enough to include, I've added the option (and a disabled block to the example conf file that we ship).

**Review by oetiker — DISMISSED** [review]

**Review by citrus-it — APPROVED** [review]

**Pull request merged**